bob-backend/src/pearl/disk_controller.rs

```diff
@@ -290,35 +290,34 @@ impl DiskController {
         Ok(groups)
     }
 
-    async fn find_group(&self, operation: &Operation) -> BackendResult<Group> {
-        Self::find_in_groups(self.groups.read().await.iter(), operation).ok_or_else(|| {
-            Error::failed(format!("cannot find actual alien folder. {:?}", operation))
-        })
+    async fn find_groups(&self, operation: &Operation) -> BackendResult<Vec<Group>> {
+        Ok(Self::find_in_groups(self.groups.read().await.iter(), operation))
     }
 
     fn find_in_groups<'pearl, 'op>(
```

Comment:
Split this function into 2: `find_all_groups` - returns vector, `find_single_group` - returns single result (in debug additionally checks, that there are no more matches in the list)

Reply:
Debug assert will require to update calls to `find_single_group`, since it requires Iterator, so I'll leave it out